File: gachisa-backend/src/main/java/com/gachisa/payment/service/PgRetryPolicy.java
package com.gachisa.payment.service;

import java.time.LocalDateTime;
import org.springframework.stereotype.Component;

@Component
public class PgRetryPolicy {

    public static final int MAX_RETRY_COUNT = 3;

    public LocalDateTime nextRetryAt(LocalDateTime now, int retryCount) {
        if (retryCount <= 1) {
            return now.plusSeconds(10);
        }
        if (retryCount == 2) {
            return now.plusSeconds(30);
        }
        return now.plusSeconds(60);
    }

    public boolean isExhausted(int retryCount) {
        return retryCount >= MAX_RETRY_COUNT;
    }
}

File: gachisa-backend/src/main/kotlin/com/gachisa/payment/service/PgRetryPolicy.kt
package com.gachisa.payment.service

import org.springframework.stereotype.Component
import java.time.LocalDateTime

@Component
class PgRetryPolicy {
    companion object {
        const val MAX_RETRY_COUNT = 3
    }

    fun nextRetryAt(now: LocalDateTime, retryCount: Int): LocalDateTime = when {
        retryCount <= 1 -> now.plusSeconds(10)
        retryCount == 2 -> now.plusSeconds(30)
        else -> now.plusSeconds(60)
    }

    fun isExhausted(retryCount: Int): Boolean = retryCount >= MAX_RETRY_COUNT
}
